docs/contributing/developing.rst

```diff
@@ -51,7 +45,20 @@ Compile the assets:
 
     $ npm run build
 
-Any Wagtail sites you start up in this virtualenv will now run against this development instance of Wagtail.  We recommend using the `Wagtail Bakery demo site <https://github.com/wagtail/bakerydemo/>`_ as a basis for developing Wagtail. Keep in mind that the setup steps for a Wagtail site may include installing a release version of Wagtail, which will override the development version you've just set up. In this case, you should install the site before running the ``pip install -e`` step, or re-run that step after the site is installed.
+You will then need to link your local Wagtail codebase to a pre-existing project you may have using Wagtail.
```

Comment:
This sentence seems a bit out of place here. We haven't explained what we mean by 'linking' - in reality it just means installing into the same virtualenv - and it's not obvious that the following `pip install` commands are the steps for doing that. It's a bit like saying "Instructions for frobnicating your flux capacitor: first, you should frobnicate your flux capacitor. Then, run the `frobnicate` command..." :-)

If the intent of this rewrite is to guide people through the difference between "running a Wagtail site with the Wagtail package code buried somewhere in `site-packages`" vs "running a Wagtail site against a working development copy of Wagtail", then I think it's counter-productive to talk about 'linking', as that's introducing new jargon that obscures the concepts they know already. I think the most helpful thing we can do is move the mention of bakerydemo to the top of the instructions, so we don't have the awkward situation where they've just built Wagtail and now need to follow the bakerydemo installation instructions while making sure they don't override that Wagtail codebase with a packaged version. The wording could be something like:

> When working on Wagtail's code, you'll need to have a running Wagtail project to test your changes against. We recommend using the Wagtail Bakery demo site for this, but it can be any pre-existing project. Once you have this installed and running as normal inside a virtual environment, we can go ahead and set up a working copy of the Wagtail codebase to be used instead of the pre-packaged Wagtail version. _(now continue with the node / git setup steps...)_